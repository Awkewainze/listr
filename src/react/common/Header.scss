@import '../../variables';

.Header {
  display: flex;
  padding: 12px;
  height: $header-height;
  box-sizing: border-box;
  background-color: $app-yellow;

  .Header__logoWrap {
    margin: 0;
  }

  .Header__logo {
    width: 48px;
  }

  .Header__logoText {
    display: none;
    margin: 0 0 0 6px;
    color: $app-black;
    font-weight: bold;
    font-size: 26px;

    @media(min-width: 640px) {
      display: block;
    }
  }

  .Header__version {
    color: $app-black;
    margin-left: 6px;
  }

  .Header__left {
    margin: 0;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .Header__right {
    margin: auto 0 auto auto;
  }
}