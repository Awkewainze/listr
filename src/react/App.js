import React from 'react';
import './App.scss';
import Header from './common/Header';
import Footer from './common/Footer';
import { Switch, Route } from 'react-router-dom';
import ActiveListContainer from './list/ActiveListContainer';

function App() {
  return (
    <div className="App">
      <Header/>
      <Switch>
        <Route path="/" component={ActiveListContainer} />
      </Switch>
      <Footer/>
    </div>
  );
}

export default App;
