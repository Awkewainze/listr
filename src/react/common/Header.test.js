import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

jest.mock('../../../package.json', () => ({
  version: '1.2.3'
}));

it('matches snapshot', () => {
  const { container } = render(<Header />);
  expect(container.firstChild).toMatchSnapshot();
});