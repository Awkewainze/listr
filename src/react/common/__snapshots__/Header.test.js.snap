// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  class="Header"
>
  <div
    class="Header__left"
  >
    <figure
      class="Header__logoWrap"
    >
      <img
        alt="Listr Logo"
        class="Header__logo"
        src="logo.svg"
      />
    </figure>
    <h1
      class="Header__logoText"
    >
      Listr
    </h1>
    <small
      class="Header__version"
    >
      v
      1.2.3
    </small>
  </div>
</header>
`;
